glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -43,6 +43,9 @@ open class GleanInternalAPI internal constructor () {
 
     private lateinit var gleanDataDir: File
 
+    // Keep track of this flag before Glean is initialized
+    private var uploadEnabled: Boolean? = null
```

Comment:
Instead of making this a nullable, can we make it `private var uploadEnabled: Boolean = true`? The default value is merely for consistency with glean-ac. We don't need this to be nullable, given the suggested changes below.